pandas/tseries/resample.py

```diff
@@ -85,9 +85,14 @@ def __init__(self, obj, groupby=None, axis=0, kind=None, **kwargs):
         self.exclusions = set()
         self.binner = None
         self.grouper = None
+        self.from_selection = False
 
         if self.groupby is not None:
-            self.groupby._set_grouper(self._convert_obj(obj), sort=True)
+            # bookeeping to disallow upsampling if not resampling on index
+            self.from_selection = (self.groupby.key is not None or
```

Comment:
I'm with you on the period stuff, but are you sure about this?  If I pull this out, it's going to break error reporting.

with:
```
In [4]: df.resample('12H', on='date').asfreq()
NotImplementedError: Upsampling from level= or on= selection is not supported, use .set_index(...) to explicitly set index to datetime-like
```

without:
```
In [5]: df.resample('12H', on='date').asfreq()
<unhelpful traceback>
ValueError: cannot include dtype 'M' in a buffer
```